pkg/noun/events.c

```diff
@@ -181,12 +181,45 @@ u3e_check(c3_c* cap_c)
 static inline c3_i
 _ce_flaw_protect(c3_w pag_w)
 {
+  // NB: if this is not static, sometimes we SIGSEGV in the memcpy,
+  // and I don't know why.
+  //
+  static c3_y con_y[16384];
+
+  // a guard page is PROT_NONE, so the memcpy will fail.  even if we
+  // excluded the current guard page, sometimes the guard page is moved
+  // without resetting PROT_NONE.  we handle all such cases by
+  // mprotecting ahead of the memcpy.
+  //
```

Comment:
> sometimes the guard page is moved without resetting PROT_NONE

That should no longer be the case, have you seen otherwise?

Reply:
I did see this on page 0, possibly related to the "guard on remap" case, which I ended up deleting anyway.  That case started getting triggered because I changed `old_w` to not be 0 on start, so that we map everything to the ephemeral file immediately.  Could just explicitly mmap everything to the ephemeral file in `u3e_live` instead of changing that logic, though.

The guard on remap case doesn't make sense to me, can you remember why it was there?

Reply:
The guard page can be anywhere between the current home-road boundaries (cap and hat). When the heap shrinks, we remap an anonymous/ephemeral segment off the end of the home heap, over all pages between our current boundary (hat) and the previously persisted boundary. This remapping is to make sure that we never SIGBUS (due to touching a page mapped off the end of the now truncated `north.bin`). But we have no guarantee that the guard page wasn't placed somewhere in that segment, so we need to check and re-protect it if necessary.

Reply:
Ah, right.  Then I should re-add that check.  I wonder if the issue is that `gar_w` defaults to 0 and isn't set until the end of `u3e_live`.  If so, there are several ways around the issue, including an explicit `mmap` in u3e_live to initialize the ephemeral mapping (analagous to the one in u3m_init).

Reply:
That seems to work in initial testing: 5d22c38d5f9f8e8cfb3624ae6231e0aa6a7de0f6